packages/expo-permissions/ios/EXPermissions/EXPermissions.m

```diff
@@ -28,266 +27,209 @@ @implementation EXPermissions
+  
+  id<UMPermissionsRequester> userNotificationRequester = [[EXUserNotificationRequester alloc] initWithNotificationProxy:[moduleRegistry  getModuleImplementingProtocol:@protocol(UMUserNotificationCenterProxyInterface)] withMetodQueqe:self.methodQueue];
+
+  [self registerRequesters:@[userNotificationRequester]];
 }
 
 # pragma mark - Exported methods
 
 UM_EXPORT_METHOD_AS(getAsync,
-                    getPermissionsWithTypes:(NSArray<NSString *> *)permissionsTypes
+                    getPermissionWithType:(NSString *)permissionType
                     resolver:(UMPromiseResolveBlock)resolve
                     rejecter:(UMPromiseRejectBlock)reject)
 {
-  NSMutableDictionary *permissions = [NSMutableDictionary new];
-  for (NSString *permissionType in permissionsTypes) {
-    NSMutableDictionary *permission = [NSMutableDictionary dictionaryWithDictionary:[self getPermissionsForResource:permissionType]];
-    // permission type not found - reject immediately
-    if (permission == nil) {
-      return reject(@"E_PERMISSIONS_UNKNOWN", [NSString stringWithFormat:@"Unrecognized permission: %@", permissionType], nil);
-    }
-
-    permissions[permissionType] = permission;
+  id<UMPermissionsRequester> requester = [self getPermissionRequesterForType:permissionType];
+  if (requester == nil) {
+    return reject(@"E_PERMISSIONS_UNKNOWN", [NSString stringWithFormat:@"Unrecognized permission: %@", permissionType], nil);
   }
-  resolve(permissions);
+  [self getPermissionUsingRequesterClass:[requester class]
+                              withResult:resolve
+                            withRejecter:reject];
 }
 
 UM_EXPORT_METHOD_AS(askAsync,
-                    askForPermissionsWithTypes:(NSArray<NSString *> *)permissionsTypes
+                    askAsyncForPermission:(NSString *)permissionType
                     resolver:(UMPromiseResolveBlock)resolve
                     rejecter:(UMPromiseRejectBlock)reject)
 {
-  [self askForPermissionsWithTypes:permissionsTypes
-                       withResults:resolve
-                      withRejecter:reject];
+  id<UMPermissionsRequester> requester = [self getPermissionRequesterForType:permissionType];
+  if (requester == nil) {
+    return reject(@"E_PERMISSIONS_UNKNOWN", [NSString stringWithFormat:@"Unrecognized permission: %@", permissionType], nil);
+  }
+  [self askForPermissionUsingRequesterClass:[requester class]
+                                 withResult:resolve
+                               withRejecter:reject];
 }
 
 # pragma mark - permission requsters / getters
 
-- (void)askForGlobalPermissions:(NSArray<NSString *> *)permissionsTypes
-                   withResolver:(void (^)(NSDictionary *))resolver
-                   withRejecter:(UMPromiseRejectBlock)reject
+
+- (void)getPermissionUsingRequesterClass:(Class)requesterClass
+                              withResult:(UMPromiseResolveBlock)onResult
+                            withRejecter:(UMPromiseRejectBlock)reject
```

Comment:
```suggestion
                              resolver:(UMPromiseResolveBlock)resolve
                           rejecter:(UMPromiseRejectBlock)reject
```
these are the conventions in the rest of the expo modules, not sure why they were different in this one but let's stick with the common conventions.